pages/admin/programs/index.tsx

```diff
@@ -0,0 +1,183 @@
+                        </HStack>
+                        <Spacer />
+                        <HStack spacing={4}>
+                            {Links.map((linkInfo) => (
+                                <NavLink
+                                    key={linkInfo.name}
+                                    href={linkInfo.url}
+                                >
+                                    {linkInfo.name}
+                                </NavLink>
+                            ))}
+                        </HStack>
+                    </Flex>
+                </Box>
+                <Divider
+                    orientation="horizontal"
+                    marginTop="25px"
+                    marginBottom="30px"
+                    border="2px"
+                />
+
+                <Box px="50px">
+                    <Text fontSize="16px">Browse Programs</Text>
+                    <InputGroup mt="25px">
+                        <InputLeftElement
+                            pointerEvents="none"
+                            children={<SearchIcon color="gray.300" />}
+                        />
+                        <Input
+                            pl={8}
+                            placeholder={"Search Programs"}
+                            onChange={(e) => {
+                                setSearchTerm(e.target.value);
+                            }}
+                        />
+                    </InputGroup>
+                </Box>
+
+                <Box ml="50px" mt="25px">
+                    {filteredCards && filteredCards.length > 0 ? (
+                        <Grid templateColumns="repeat(3, 1fr)" gap={6}>
+                            {" "}
+                            {filteredCards.map((item, idx) => {
+                                return (
+                                    <GridItem key={idx}>
+                                        <BrowseProgramCard cardInfo={item} />
+                                    </GridItem>
+                                );
+                            })}
+                        </Grid>
+                    ) : (
+                        <Box pr="50px">
+                            <AdminEmptyState
+                                w="100"
+                                h="100%"
+                                isLoading={isLoading}
+                            >
+                                No filtered classes available
+                            </AdminEmptyState>
+                        </Box>
+                    )}
+                </Box>
+            </Box>
+        </Wrapper>
+    );
+};
+
+export default BrowsePrograms;
+
+/**
+ * getServerSideProps gets the session before this page is rendered
+ */
+export const getServerSideProps: GetServerSideProps = async (context) => {
+    // obtain the next auth session
+    const session = await getSession(context);
+    // TODO : check session for admin/volunteer , if not redirect to no access
```

Comment:
something like this works for now. I'll clean this up later with a helper method.

```suggestion
    if (!session) {
        return {
            redirect: {
                destination: "/login",
                permanent: false,
            },
        };
    } else if (
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        ![roles.PROGRAM_ADMIN, roles.TEACHER].includes((session as any).role)
    ) {
        return {
            redirect: {
                destination: "/no-access",
                permanent: false,
            },
        };
    }
```